### PR TITLE
virtual & virtual-attachments documentation updates

### PR DESCRIPTION
**Pull request by vmo69**
DOV-4590

### CODE REVIEW

**Comment by cmouse** [comment]
should we indicate that this a pro-only feature?

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Then, you'll have to create a :ref:`namespaces` for the virtual-attachments,
+for example:
+
+.. code-block:: none
+
+  namespace {
+    prefix = virtual-attachments/
+    separator = /
+    location = virtual:~/virtual-attachments
```
Comment (cmouse) [inline source/configuration_manual/virtual_attachments_plugin.rst:28]: This is not correct.

Comment (cmouse) [reply source/configuration_manual/virtual_attachments_plugin.rst:28]: also it's missing the virtual plugin config

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+For every mailbox that shall get its own attachments folder, an according folder
+below that namespace needs to be created. The mailbox names need to match
+existing real or virtual mailboxes. Those will be mirrored below the
+virtual-attachments namespace then and contain virtual messages - one per
+attachment - based on the referenced original mailbox. The mailboxes can be
+auto created.
+
```
Comment (cmouse) [inline source/configuration_manual/virtual_attachments_plugin.rst:37]: the above paragraph could use bit work, it's very hard to read now.

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+Storage location with obox
+--------------------------
+
+When using the virtual-attachments plugin with obox, the virtual-attachments
```
Comment (cmouse) [inline source/configuration_manual/virtual_attachments_plugin.rst:41]: link to obox.

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+indexes folder must be configured to be placed in the users home directory
+under the foldername "virtual-attachments" otherwise the indexes & caches will
+be lost when cleaning metacache.
+
+.. code-block:: none
+
+        location = virtual:~/virtual-attachments
```
Comment (cmouse) [inline source/configuration_manual/virtual_attachments_plugin.rst:48]: this is again wrong.

File: source/configuration_manual/virtual_plugin.rst [hunk 5]
```diff
@@ -35,6 +35,21 @@ default it uses the `fs` layout, so you can create directories such as:
...
+Storage location with obox
+--------------------------
+
+When using the virtual plugin with obox, the virtual indexes folder must be
+configured to be placed in the users home directory under the foldername
+"virtual" otherwise the indexes will be lost when cleaning metacache.
+
+.. code-block:: none
+
+        location = virtual:/etc/dovecot/virtual:INDEX=~/virtual
```
Comment (cmouse) [inline source/configuration_manual/virtual_plugin.rst:47]: and this too.

File: source/configuration_manual/virtual_attachments_plugin.rst (hunk 1 above)
Comment (cmouse) [reply source/configuration_manual/virtual_attachments_plugin.rst:28]: the location should be attachments =)

**Review by cmouse — CHANGES_REQUESTED** [review]

File: source/configuration_manual/virtual_attachments_plugin.rst (hunk 2 above)
Comment (vmo69) [reply source/configuration_manual/virtual_attachments_plugin.rst:45]: Better now?

File: source/configuration_manual/virtual_plugin.rst (hunk 5 above)
Comment (vmo69) [reply source/configuration_manual/virtual_plugin.rst:47]: Here it's correct i think as it's in the virtual plugin not the virtual-attachments.

File: source/configuration_manual/virtual_plugin.rst [hunk 6]
```diff
@@ -35,6 +35,21 @@ default it uses the `fs` layout, so you can create directories such as:
 If you prefer to use Maildir++ layout instead, you can simply append
 ``:LAYOUT=maildir++`` to the location.
 
+Storage location with obox
+--------------------------
+
+When using the virtual plugin with obox, the virtual indexes folder must be
+configured to be placed in the users home directory under the foldername
```
Comment (sirainen) [inline source/configuration_manual/virtual_plugin.rst:42]: How about:

```
.. versionadded:: v2.3.17

When using the virtual plugin with obox, the virtual INDEX location must point
to a directory named "virtual" in the user home directory. This way the virtual indexes
are added to the obox root index bundles and will be preserved when user moves between
backends or when metacache is cleaned.
```

File: source/configuration_manual/virtual_plugin.rst [hunk 7]
```diff
@@ -35,6 +35,21 @@ default it uses the `fs` layout, so you can create directories such as:
...
+"virtual" otherwise the indexes will be lost when cleaning metacache.
+
+.. code-block:: none
+
+        location = virtual:/etc/dovecot/virtual:INDEX=~/virtual
+
+The virtual indexes will be stored in the user root bundle.
+
+.. versionadded:: 2.3.17
```
Comment (sirainen) [inline source/configuration_manual/virtual_plugin.rst:51]: Usually the versionadded is in the beginning of the sections I think.

Also it's missing the setting to disable this behavior. I guess that should be also in the settings/ directory.

File: source/configuration_manual/virtual_attachments_plugin.rst (hunk 2 above)
Comment (sirainen) [reply source/configuration_manual/virtual_attachments_plugin.rst:46]: I think it's good, but could use an example at the bottom, like:

 * Attachments in INBOX are in virtual-attachments/INBOX
 * Attachments in virtual/All are in virtual-attachments/virtual/All

Also isn't there often the virtual/all autocreated? I'm not sure actually. But I think:

```
namespace {
  mailbox virtual/All {
    auto = create
  }
}
```

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+mailbox. The virtual-attachments mailboxes can be auto created
+(see :ref:`mailbox_settings`).
+
+Storage location with obox
+--------------------------
+
+When using the virtual-attachments plugin with :ref:`obox <obox_settings>`,
+the virtual-attachments indexes folder must be configured to be placed in the
+users home directory under the foldername "virtual-attachments" otherwise the
+indexes & caches will be lost when cleaning metacache.
```
Comment (sirainen) [inline source/configuration_manual/virtual_attachments_plugin.rst:53]: This could be synced with the virtual's suggested changes.

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/virtual_attachments_plugin.rst (hunk 2 above)
Comment (vmo69) [reply source/configuration_manual/virtual_attachments_plugin.rst:59]: Ok i added the examples, namespace section names and autocreated "All"  mailboxes.

File: source/configuration_manual/virtual_plugin.rst (hunk 7 above)
Comment (vmo69) [reply source/configuration_manual/virtual_plugin.rst]: I added it under obox/advanced

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+.. code-block:: none
+
+  namespace virtual {
+    prefix = virtual/
+    separator = /
+    location = virtual:/etc/dovecot/virtual:INDEX=~/virtual
+
+    mailbox All {
+        auto = create
+    }
```
Comment (sirainen) [inline source/configuration_manual/virtual_attachments_plugin.rst:34]: This isn't needed / won't do anything useful. The All mailbox needs to be explicitly configured in filesystem.

File: source/configuration_manual/virtual_attachments_plugin.rst [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  namespace virtual-attachments {
+    prefix = virtual-attachments/
+    separator = /
+    location = attachments:~/virtual-attachments
+
+    mailbox virtual/All {
+        auto = create
```
Comment (sirainen) [inline source/configuration_manual/virtual_attachments_plugin.rst:43]: good, but too much indentation. 2 spaces is the standard in config.

File: source/configuration_manual/virtual_plugin.rst [hunk 11]
```diff
@@ -35,20 +35,26 @@ default it uses the `fs` layout, so you can create directories such as:
...
+.. _virtual_plugin_obox_secondary_indexes:
+
 Storage location with obox
 --------------------------
 
-When using the virtual plugin with obox, the virtual indexes folder must be
-configured to be placed in the users home directory under the foldername
-"virtual" otherwise the indexes will be lost when cleaning metacache.
+.. versionadded:: v2.3.17
+
```
Comment (sirainen) [inline source/configuration_manual/virtual_plugin.rst:44]: These changes should be in the previous commit.

**Review by sirainen — CHANGES_REQUESTED** [review]

**Review by sirainen — APPROVED** [review]

**Review by cmouse — APPROVED** [review]

**Pull request merged**